src/test/kotlin/uk/gov/justice/digital/hmpps/bodyscan/service/LimitStatusServiceTest.kt

```diff
@@ -0,0 +1,81 @@
+package uk.gov.justice.digital.hmpps.bodyscan.service
+
+import org.assertj.core.api.Assertions
+import org.junit.jupiter.api.Test
+import org.mockito.kotlin.any
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+import uk.gov.justice.digital.hmpps.bodyscan.apiclient.BodyScanApiClient
+import uk.gov.justice.digital.hmpps.bodyscan.apiclient.model.PersonalCareCounter
+import uk.gov.justice.digital.hmpps.bodyscan.model.LimitStatusResponse
+import java.time.LocalDate
+import java.time.Year
+
+class LimitStatusServiceTest {
+  private val prisonApiClient: BodyScanApiClient = mock()
+  private val limitStatusService = LimitStatusService(prisonApiClient)
+
+  @Test
+  fun `get limit status for year and prison numbers`() {
+    val prisonNumbers = listOf("G8266VG", "G8874VT", "G8874VZ")
+    val year = Year.of(2022)
+    whenever(prisonApiClient.getPersonalCareNeedsForPrisonNumbers(any(), any(), any(), any())).thenReturn(
+      listOf(
+        PersonalCareCounter("G8266VG", 1),
+        PersonalCareCounter("G8874VT", 120),
+        PersonalCareCounter("G8874VZ", 160)
```

Comment:
it's good practice to test around the boundaries around levels -  if you have a limit of 120 then you would test 119, 120, and 121. 

This highlights fence post errors and inclusivity issues.  

(our limit is 116)

Reply:
sorted